internal/socket/tcp_socket.go

```diff
@@ -123,6 +123,10 @@ func tcpSocket(proto, addr string, passive bool, sockOpts ...Option) (fd int, ne
 		return
 	}
 	defer func() {
+		// ignore EINPROGRESS for non-blocking socket connect, should be processed by caller
+		if t, ok := err.(*os.SyscallError); ok && t.Err == unix.EINPROGRESS {
```

Comment:
move this into the `if err != nil` block.

Reply:
connect/bind error is wrapped with `SyscallError` causing that the err is mixed with `syscall.Errno` and  `SyscallError`. so here should unwrap the err.